src/sfsecrethelper.h

```diff
@@ -0,0 +1,44 @@
+#ifndef SFSECRETHELPER_H
+#define SFSECRETHELPER_H
+
+#include <QObject>
+#include <QtCore/QObject>
+#include <QtCore/QByteArray>
+
+#include <Secrets/secretmanager.h>
+#include <Secrets/secret.h>
+
+class QQmlEngine;
+class QJSEngine;
+
+class SFSecretHelper : public QObject
+{
+    Q_OBJECT
+public:
+
+    struct Response {
+        int returncode;
+        QByteArray data;
+    };
+
+    explicit SFSecretHelper(QString collection, QObject *parent = nullptr);
+
+    static void instantiate(const QString &collection, QObject *parent = nullptr);
+    static SFSecretHelper &getInstance();
+    static QObject *provider(QQmlEngine *engine, QJSEngine *scriptEngine);
+
+    Response store(const QString &key, const QByteArray &value);
+    Response get(const QString &key);
+    Response remove(const QString &key);
```

Comment:
I considered this, but I'm conviced this is very confusing because depending which method was called, a QByteArray is either returned on success or failure(to eventually still handle cases when SFSecrets fails, I haven't implemented anything to this yet.). Additionally, it's faster if you only need to check for an int.

Reply:
@llewelld  Could you please make a comment how I should proceed on this?

Reply:
Okay, in this case I don't have any problem with leaving this as is. [The Internet](https://stackoverflow.com/questions/33736172/does-qt-have-its-own-boostoptional-alternative) seems to think that using a `QVariant` return type is an alternative to `std::optional`, but i don't see this as being preferable to what you have here, which I agree certainly makes the semantics clearer.

Reply:
I also though about writing an own simple implementation of std::optional, but it doesn't really make sense.